cf/domain.py

```diff
@@ -156,6 +158,79 @@ def close(self):
             removed_at="5.0.0",
         )  # pragma: no cover
 
+    @classmethod
+    def create(cls, x, y, dx, dy):
+        '''
+        Create a new domain with the given longitude and latitude bounds and intervals.
+        
+        :Parameters:
+        
+            x: tuple
+                A tuple of two numeric values representing the lower and upper bounds of the longitude dimension.
+        
+            y: tuple
+                A tuple of two numeric values representing the lower and upper bounds of the latitude dimension.
+        
+            dx: float
+                The interval between points along the longitude axis.
+        
+            dy: float
+                The interval between points along the latitude axis.
+        
+        :Returns:
+        
+            `Domain`
+                The newly created domain with the specified longitude and latitude bounds, domain axes, and coordinate axes.
+        
+        **Examples:**
+        
+        >>> import cf
+        >>> domain = cf.Domain.create((-180, 180), (-90, 90), 1, 1)
+        >>> print(domain)
+        Dimension coords: longitude(361) = [-180, ..., 180] degrees_east
+                : latitude(181) = [-90, ..., 90] degrees_north
+
+        '''
+
+        if not (len(x) == 2 and len(y) == 2):
+            raise ValueError("x and y must have two elements each.")
+
+        x_range = x[1] - x[0]
+        y_range = y[1] - y[0]
+
+        if not (x_range % dx == 0 and y_range % dy == 0):
+            raise ValueError("dx and dy must be divisible by the respective ranges of x and y.")
+
+        domain = cls()
+
+        x_points = numpy.arange(x[0], x[1] + dx, dx)
+        y_points = numpy.arange(y[0], y[1] + dy, dy)
+
+        # Create domain axes for longitude and latitude
+        domain_axis_longitude = domain.set_construct(domain._DomainAxis(len(x_points)))
+        domain_axis_latitude = domain.set_construct(domain._DomainAxis(len(y_points)))
```

Comment:
```suggestion
        domain_axis_longitude = domain.set_construct(domain._DomainAxis(x_points.size))
        domain_axis_latitude = domain.set_construct(domain._DomainAxis(y_points.size))
```
`.size` is faster :)